### PR TITLE
Clarify what gets printed in the generated quantities block

### PR DESCRIPTION
**Pull request by mcol**
#### Submission Checklist

- [X] Builds locally 
- [X] Declare copyright holder and open-source license: see below

#### Summary
Incorporates a suggestions from https://discourse.mc-stan.org/t/do-not-output-a-variable-from-generated-quantities/13228/5. If it can be further improved I'm happy to amend.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Marco Colombo

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/reference-manual/blocks.Rmd [hunk 1]
```diff
@@ -565,4 +565,13 @@ After the generated quantities statements are executed, the constraints
 on the declared generated quantity variables are validated.
 
 All variables declared as generated quantities are printed as part of
```
Comment (bob-carpenter) [inline src/reference-manual/blocks.Rmd]: Technically, what I wrote is correct.  Variables declared in nested blocks are local variables, not generated quantities.  Only the first variables declared at the top are considered generated quantities.  

I would rather rephrase this to add another sentnece that says that variables declared in local scopes are local variables and not generated quantities and thus won't be printed.  Same for transformed parameters and transformed data---in general, local variables are not visible beyond the block in which they're declared, which means they're not visible for output.
But I'll just merge this if it was confusing before as nobody's going to get confused by the semantic implication that

**Comment by mcol** [comment]
Is this what you meant?
